app/hooks/useContracts.ts

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from "react";
+
+export function useContracts({ network }: { network: string }) {
+  const [
+    { xMetricJson, questionAPIJson, questionStateController, bountyQuestionJson, costController, vaultJson },
+    setContracts,
+  ] = useState(getContracts(network));
+
+  useEffect(() => {
+    setContracts(getContracts(network));
+  }, [network]);
+
+  return {
+    xMetricJson,
+    questionAPIJson,
+    questionStateController,
+    bountyQuestionJson,
+    costController,
+    vaultJson,
+  };
+}
+
+function getContracts(network: string) {
```

Comment:
I'm confused I don't think require works on the front end - this should be a back end call in useLoader state